library/systemd/src/lib/yast2/systemd_unit.rb

```diff
@@ -37,7 +37,23 @@ class SystemdUnit
     SUPPORTED_TYPES  = %w(service socket target).freeze
     SUPPORTED_STATES = %w(enabled disabled).freeze
 
-    DEFAULT_PROPERTIES = {
+    # Values of {#active_state} fow which we consider a unit "active".
+    #
+    # systemctl.c:check_unit_active uses (active, reloading)
+    # For bsc#884756 we also consider "activating" to be active.
+    #
+    # (The remaining states are "deactivating", "inactive", "failed".)
+    ACTIVE_STATES = ["active", "activating", "reloading"].freeze
```

Comment:
that is a bit problematic, as systemd can add more states and we had such problem in past, that is why we call explicit systemd command is_active that move responsibility to systemd to say if it is active.

Reply:
> that is why we call explicit systemd command is_active

And that is why our module was so f**king slow. Should I add a comment like this?
> 10 to 50ms per call of `systemctl is-active` * 100 to 300 services (depending on hardware and software installed, VM or not) is 1 to 15 seconds delay. That is why we try really hard to avoid many systemctl calls.

Reply:
yes, comment would be great to explain it. Also what I have in mind is explicit check that it is known state ( written only in comment now ) and if not, at least log error ( exception is maybe too strong, but who knows ).

Reply:
Added